docker-wrappers/SPRAS/README.md

```diff
@@ -48,16 +48,66 @@ DOCKER_DEFAULT_PLATFORM=linux/amd64 docker build -t reedcompbio/spras:v0.1.0 -f
+is to use the the snakemake HTCondor executor to parallelize the workflow by submitting each job to its own EP.
+
+### Submitting All Jobs to a Single EP
+
+Navigate to the `spras/docker-wrappers/SPRAS` directory and create the `logs/` directory. Then run `condor_submit spras.sub`, which will submit SPRAS
+to HTCondor as a single job with as many cores as indicated by the `NUM_PROCS` line in `spras.sub`, using the value of `EXAMPLE_CONFIG` as the SPRAS
+configuration file. Note that you can alter the configuration file to test various workflows, but you should leave `unpack_singularity = true`, or it
+is likely the job will be unsuccessful. By default, the `example_config.yaml` runs everything except for `cytoscape`, which appears to fail periodically
+in HTCondor.
+
+**Note**: The `spras.sub` submit file is an example of how this workflow could be submitted from a CHTC Access Point (AP) to the OSPool. To run in the local
+CHTC pool, omit the `+WantGlideIn` and `requirements` lines
+
+### Submitting Parallel Jobs
+
+Parallelizing SPRAS workflows with HTCondor currently requires an experimental executor for HTCondor that has been forked from the upstream [HTCondor Snakemake executor](https://github.com/jhiemstrawisc/snakemake-executor-plugin-htcondor/tree/spras-feature-dev).
+To get this executor, clone the forked repository using the following:
+```bash
+git clone -b spras-feature-dev https://github.com/jhiemstrawisc/snakemake-executor-plugin-htcondor.git
+```
+
+Then, from your activated `spras` conda environment (important), run:
+```bash
+pip install snakemake-executor-plugin-htcondor/
+```
+
+Currently, this executor requires that all input to the workflow is scoped to the current working directory. Therefore, you'll need to copy the
+Snakefile and your input directory (as specified by `example_config.yaml`) to this directory:
+```bash
+cp ../../Snakefile . && \
+cp -r ../../input .
+```
+
+It's also necessary for this workflow to create an Apptainer image from the published Docker image. See [Creating an Apptainer image for SPRAS](#creating-an-apptainer-image-for-spras)
+for instructions.
+
+To start the workflow with HTCondor in the CHTC pool, run:
+```bash
+snakemake --profile spras_profile
+```
+
+Resource requirements can be adjusted as needed in `spras_profile/config.yaml`, and HTCondor logs for this workflow can be found in `.snakemake/htcondor`.
+You can set a different log directory by adding `htcondor-jobdir: /path/to/dir` to the profile's configuration.
 
+To run this same workflow in the OSPool, add the following to the profile's default-resources block:
+```
+  classad_WantGlideIn: true
+  requirements: |
+    '(HAS_SINGULARITY == True) && (Poolname =!= "CHTC")'
+```
+
+**Note**: This workflow requires that the terminal session responsible for running snakemake stays active. Closing the terminal will suspend jobs,
```

Comment:
This would be an issue for full workflows with thousands of jobs. We may want to talk to CHTC facilitators about preferred ways to run this. I'm assuming they would not want us to use tmux or screen to keep an active session, but I'm not sure what is preferable. Probably something that runs the Snakemake process with HTCondor itself. That seems too complicated.

Reply:
I've given some thought to this, and I believe the answer is to wrap the workflow in an HTCondor "local universe" job. Local universe allows you to run an executable from the AP as an HTCondor job. I'd like to make implementation of the wrapper HTCondor job a second PR, because it adds a few layers of complexity and feels like scope creep for this PR.

Reply:
Agreed that we'll make this a follow up pull request